packages/core/src/hydration/event_replay.ts

```diff
@@ -58,11 +57,18 @@ export function withEventReplay(): Provider[] {
     {
       provide: ENVIRONMENT_INITIALIZER,
       useValue: () => {
-        setDisableEventReplayImpl((el: RElement) => {
-          if (el.hasAttribute(JSACTION_ATTRIBUTE)) {
+        setDisableEventReplayImpl((rEl, eventName, listenerFn) => {
+          if (rEl.hasAttribute(JSACTION_ATTRIBUTE)) {
+            const el = unwrapRNode(rEl) as Element;
             // We don't immediately remove the attribute here because
             // we need it for replay that happens after hydration.
-            removeJsactionQueue.push(el);
+            if (!jsactionMap.has(el)) {
+              jsactionMap.set(el, new Map());
+            }
+            if (!jsactionMap.get(el)!.has(eventName)) {
+              jsactionMap.get(el)!.set(eventName, []);
+            }
+            jsactionMap.get(el)!.get(eventName)!.push(listenerFn);
```

Comment:
nit: minor refactoring to avoid reading the same key from the map:
```suggestion
            const eventMap = jsactionMap.get(el)!;
            if (!eventMap.has(eventName)) {
              eventMap.set(eventName, []);
            }
            eventMap.get(eventName)!.push(listenerFn);
```

Reply:
done